src/app/client/js/es6/controllers.js

```diff
@@ -233,26 +367,86 @@
+          return showCtfChoices;
+        }
+        if (parentFamilyName === "VSTS") {
+          return showVSTSChoices;
+        }
+       return false;
+      }
 
       $scope.familyHasBeenSelected = () => family !== '';
 
-      $scope.familyIsSelectedIcon = familyName =>{
-        return (family === familyName || familyHover === familyName) ? `icon-${familyName.toLowerCase()}-selected-32x32.png` : `icon-${familyName.toLowerCase()}-nonselected-32x32.png`;
+      $scope.familyIcon = familyName => {
+        let parentFamilyName = getParentFamilyName(familyName);
+        let icon = null;
+        if ($scope.families.hasOwnProperty(parentFamilyName)) {
+          let fam = $scope.families[parentFamilyName];
+          if (fam.icon) {
+            icon = fam.icon.active;
+          }
+        }
+        if (!icon) {
+          icon = `icon-${getParentFamilyName(familyName).toLowerCase()}-selected-32x32.png`
+        }
+        return `${serviceUrl}/` + icon;
       }
 
-      $scope.vstsFamilyIsSelectedIcon = familyName =>{
-        return (familyHover === familyName || shouldBeTreatedAsVsts(family)) ? `icon-${familyName.toLowerCase()}-selected-32x32.png` : `icon-${familyName.toLowerCase()}-nonselected-32x32.png`;
+      $scope.getSubFamilyIcon = (familyName, subFamilyName) => {
+        if (!familyName) {
+          familyName = getParentFamilyName(subFamilyName);
+        }
+        let icon = '';
+        let fams = $scope.families;
+        if (fams && fams.hasOwnProperty(familyName) && fams[familyName].families) {
+          let subFams = fams[familyName].families;
+          if (subFams.hasOwnProperty(subFamilyName)) {
+            let subFam = subFams[subFamilyName];
+            if (subFam.icon && subFam.icon.active) {
+              icon = subFam.icon.active;
+            }
+          }
+        }
+        if (!icon) {
+          icon = `icon-${subFamilyName.toLowerCase()}-selected-24x24.png`;
+        }
+        return (`${serviceUrl}/` + icon);
       }
 
-      $scope.familyIcon = familyName => {
-        if(shouldBeTreatedAsVsts(familyName)) {
-          return `${serviceUrl}/icon-vsts-selected-32x32.png`;
+      $scope.getIsSelectedFamilyIcon = (familyName) => {
+        let icon = '';
+        let fams = $scope.families;
+        let isActive = (familyHover === familyName) || (getParentFamilyName(family) === familyName); 
+        if (fams && fams.hasOwnProperty(familyName)) {
+          let fam = fams[familyName];
+          if (fam.icon) {
+            if (isActive) {
+              icon = fam.icon.active;
+            }
+            else {
+              icon = fam.icon.inactive;
+            }
+          }
+        }
+        if (!icon) {
+          icon = isActive ? `icon-${familyName.toLowerCase()}-selected-32x32.png` : `icon-${familyName.toLowerCase()}-nonselected-32x32.png`;
         }
-        return `${serviceUrl}/icon-${familyName.toLowerCase()}-selected-32x32.png`;
+        return icon;
```

Comment:
Could it be an issue here that `${serviceUrl}` is not utilized?

Reply:
This function replaces 'familyIsSelectedIcon' which didn't include 'serviceUrl'. You can see the usage  on line 489 in commitStreamAdmin.html, where the return value is prefixed with serviceUrl.

Reply:
👍